src/modules/design/design.ts

```diff
@@ -40,6 +41,9 @@ export class Design {
 
   @bindable() public xmlForDiff: string;
   public diagramDetail: DiagramDetail;
+  public filteredSolutions: Array<ISolution> = [];
+  public diagramArray: Array<IDiagram | object> = [];
+  public selectedDiagram: {diagram: IDiagram, solutionName: string};
```

Comment:
Move this into a type above the class

```
type DiagramWithSolution = {
  diagram: IDiagram,
  solutionName: string
};
```

```suggestion
  public selectedDiagram: DiagramWithSolution;
```